pkg/controller/acicontainersoperator.go

```diff
@@ -311,124 +389,134 @@ func (t *OperatorHandler) CheckOwnerReference(reference []metav1.OwnerReference)
 			*metav1.NewControllerRef(acicontainersoperator, operators.SchemeGroupVersion.WithKind("AciContainersOperator")),
 		}
 
-		_, err = hostdaemonsetclient.Update(t.HostDaemonset)
+		_, err := hostdaemonsetclient.Update(c.Resources.HostDaemonset)
 		if err != nil {
 			log.Error(err.Error())
+			return false
 		}
 	}
+	log.Info("Successfully updated owner reference to the aci-container-host daemonset")
+	return true
+
+}
 
-	ovsdaemonsetclient := k8sclient.AppsV1().DaemonSets(os.Getenv("SYSTEM_NAMESPACE"))
+func (c *Controller) UpdateOvsDaemonsetOwnerReference(acicontainersoperator *operators.AciContainersOperator) bool {
+	ovsdaemonsetclient := c.K8s_Clientset.AppsV1().DaemonSets(os.Getenv("SYSTEM_NAMESPACE"))
 	if ovsdaemonsetclient == nil{
 		log.Info("Error in Fetching ovsdaemonsetclient...")
-		return
+		return false
 	}
 
-	t.OvsDaemonset, _ = hostdaemonsetclient.Get("aci-containers-openvswitch",metav1.GetOptions{})
-	if t.OvsDaemonset == nil {
+	c.Resources.OvsDaemonset, _ = ovsdaemonsetclient.Get("aci-containers-openvswitch",metav1.GetOptions{})
+	if c.Resources.OvsDaemonset == nil {
 		log.Info("aci-containers-openvswitch daemonset is nil.....returning")
-		return
+		return false
 	}
 
-	if  !t.CheckOwnerReference(t.OvsDaemonset.OwnerReferences){
-		t.OvsDaemonset.OwnerReferences = []metav1.OwnerReference{
+	if  !c.CheckOwnerReference(c.Resources.OvsDaemonset.OwnerReferences){
+		c.Resources.OvsDaemonset.OwnerReferences = []metav1.OwnerReference{
 			*metav1.NewControllerRef(acicontainersoperator, operators.SchemeGroupVersion.WithKind("AciContainersOperator")),
 		}
 
-		_, err = ovsdaemonsetclient.Update(t.OvsDaemonset)
+		_, err := ovsdaemonsetclient.Update(c.Resources.OvsDaemonset)
 		if err != nil {
 			log.Error(err.Error())
+			return false
 		}
 	}
+	log.Info("Successfully updated owner reference to the aci-containers-openvswitch daemonset")
+	return true
+}
+
+func (c *Controller) handleOperatorCreate(obj interface{}) bool {
+
+	log.Info("OperatorHandler.ObjectCreated")
+
+	acicontainersoperator := obj.(*operators.AciContainersOperator)
+
+	log.Debug(acicontainersoperator.Spec.Config)
+
+	if (acicontainersoperator.Spec.Config == ""){
+		log.Info("acicnioperator CR config is Nil")
+		return true
+	}
+	
+	dec, err := base64.StdEncoding.DecodeString(acicontainersoperator.Spec.Config)
+	if err != nil {
+		panic(err)
+		return true
+	}
+
+	f, err := os.Create("aci-deployment.yaml")
+	if err != nil {
+		panic(err)
+		return true
+	}
+	if _, err := f.Write(dec); err != nil {
+		panic(err)
+		return true
+	}
+	if err := f.Sync(); err != nil {
+		panic(err)
+		return true
+	}
+	if err := f.Close();err != nil{
+		panic(err)
+		return true
+	}
+	log.Info("Applying Aci Deployment")
+
+	//Currently the Kubectl version is v.1.14. This will be updated by the acc-provision according
+	//to the platform specification
+	cmd := exec.Command("kubectl","apply","-f","aci-deployment.yaml")
+	log.Debug(cmd)
+	_, _ = cmd.Output()
```

Comment:
Probably it is better to do retry here if you are applying cmd, check for error and do retry

Reply:
I will check for the type of error. Sometimes it's a generic message where things are applied but still, it shows a message like "please apply your changes to the latest version and try again". I will do a fix where we can handle for any other error conditions